datafusion/core/tests/sql/expr.rs

```diff
@@ -796,6 +796,11 @@ async fn test_regex_expressions() -> Result<()> {
 
 #[tokio::test]
 async fn test_cast_expressions() -> Result<()> {
+    test_expression!("CAST([1,2,3,4] AS INT[])", "[1, 2, 3, 4]");
+    test_expression!(
+        "CAST([1,2,3,4] AS NUMERIC(10,4)[])",
```

Comment:
👍 

I ran this test without the code change and it failed like

```
---- sql::expr::test_cast_expressions stdout ----
thread 'sql::expr::test_cast_expressions' panicked at 'Creating logical plan for 'SELECT CAST([1,2,3,4] AS INT[])': NotImplemented("Unsupported SQL type Array(Int(None))")', datafusion/core/tests/sql/mod.rs:538:45

```

In case anyone was curious